src/server/routes/api/index.js

```diff
@@ -18,36 +17,76 @@ function getSqsPayload(object) {
   };
 }
 
+function isSqsDataValid(params, type) {
+  const { barcodes, protectCGD, action, email } = params;
+  if (type === 'transfer') {
+    const { bibRecordNumber } = params;
 
-export function updateMetadata(req, res, next) {
-  const params = req.body;
+    if (isEmpty(barcodes) || isEmpty(bibRecordNumber) || typeof protectCGD !== 'boolean' || isEmpty(action) || isEmpty(email)) {
```

Comment:
I feel we can test `isEmpty(barcodes) || typeof protectCGD !== 'boolean' || isEmpty(action) || isEmpty(email)` together before we get into the condition check for `type` since they are necessary in both `update` and `transfer`

Reply:
that is very true, good catch -- I will simplify the logic. Thanks kTp~